src/nnet3/nnet-utils.cc

```diff
@@ -101,9 +101,10 @@ static void ComputeSimpleNnetContextForShift(
     input.indexes.push_back(Index(n, t));
     output.indexes.push_back(Index(n, t));
   }
-  // the assumption here is that the network just requires the ivector at time
-  // t=0.
-  ivector.indexes.push_back(Index(n, 0));
+  // the assumption here is that the network just requires the ivectors starting
+  // from time t=0, and the number is at most input_end.
```

Comment:
hm- actually, it may be OK, but would be better to add them with the correct period, worked out from the GetIvectorPeriod() function.